Address Safer CPP warnings in WebSWContextManagerConnection.cpp : updateAppInitiatedValue

kmonsen opened the pull request:
#### af3f2c5bde1904db41c7773cb1c4d10985d81104
<pre>
Address Safer CPP warnings in WebSWContextManagerConnection.cpp : updateAppInitiatedValue
<a href="https://bugs.webkit.org/show_bug.cgi?id=289629">https://bugs.webkit.org/show_bug.cgi?id=289629</a>
<a href="https://rdar.apple.com/146879414">rdar://146879414</a>

Reviewed by Chris Dumez.

Making member refs const when they can be.

* Source/WebKit/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
(WebKit::WebSWContextManagerConnection::updateAppInitiatedValue):
(WebKit::WebSWContextManagerConnection::installServiceWorker):
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.h:

Canonical link: <a href="https://commits.webkit.org/292294@main">https://commits.webkit.org/292294@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/abd0c3f17f61f80eb0b44ea199ee38a7f6d46fda

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/95283 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14884 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4742 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/100328 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45785 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/97331 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/15172 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/23315 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72657 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29933 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/98286 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/11327 "Found 1 new test failure: fast/forms/ios/focus-input-via-button.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/86004 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52989 "Passed tests") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/94775 "Build is in progress. Recent messages:Passed webkitpy tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/11040 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3742 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/45124 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/87955 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/81216 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3837 "Found 1 new test failure: http/tests/workers/service/registration-task-queue-scheduling-1.html (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/102366 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/93907 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/22332 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/16291 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/81654 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/94859 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22580 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/82024 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/81052 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20353 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25624 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/3021 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/15585 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/22302 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/27673 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/116595 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled JSC") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21961 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25435 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23700 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
```diff
@@ -100,7 +100,8 @@ WebSWContextManagerConnection::WebSWContextManagerConnection(Ref<IPC::Connection
 {
 #if ENABLE(CONTENT_EXTENSIONS)
-    m_userContentController->addContentRuleLists(WTFMove(initializationData.contentRuleLists));
+    Ref protectedController = m_userContentController;
```
This is not needed. `m_userContentController` never gets reassigned so we should just mark the data member as `const`

cdumez commented on Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
```diff
@@ -113,15 +114,17 @@ WebSWContextManagerConnection::~WebSWContextManagerConnection() = default;
 {
-    m_connectionToNetworkProcess->addWorkQueueMessageReceiver(Messages::WebSWContextManagerConnection::messageReceiverName(), m_queue.get(), *this);
-    m_connectionToNetworkProcess->sendWithAsyncReply(Messages::NetworkConnectionToWebProcess::EstablishSWContextConnection { m_webPageProxyID, m_site, m_serviceWorkerPageIdentifier }, WTFMove(completionHandler), 0);
+    Ref protectedConnection = m_connectionToNetworkProcess;
```
Same as above.

cdumez commented on Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
```diff
@@ -113,15 +114,17 @@ WebSWContextManagerConnection::~WebSWContextManagerConnection() = default;
     ASSERT(isMainRunLoop());
 
-    m_connectionToNetworkProcess->removeWorkQueueMessageReceiver(Messages::WebSWContextManagerConnection::messageReceiverName());
+    Ref protectedConnection = m_connectionToNetworkProcess;
```
ditto.

cdumez commented on Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
```diff
@@ -238,12 +241,14 @@ void WebSWContextManagerConnection::setUserAgent(String&& userAgent)
 void WebSWContextManagerConnection::serviceWorkerStarted(std::optional<ServiceWorkerJobDataIdentifier> jobDataIdentifier, ServiceWorkerIdentifier serviceWorkerIdentifier, bool doesHandleFetch)
 {
-    m_connectionToNetworkProcess->send(Messages::WebSWServerToContextConnection::ScriptContextStarted { jobDataIdentifier, serviceWorkerIdentifier, doesHandleFetch }, 0);
+    Ref protectedConnection = m_connectionToNetworkProcess;
```
ditto.

cdumez commented on Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
```diff
@@ -238,12 +241,14 @@ void WebSWContextManagerConnection::setUserAgent(String&& userAgent)
 void WebSWContextManagerConnection::serviceWorkerFailedToStart(std::optional<ServiceWorkerJobDataIdentifier> jobDataIdentifier, ServiceWorkerIdentifier serviceWorkerIdentifier, const String& exceptionMessage)
 {
-    m_connectionToNetworkProcess->send(Messages::WebSWServerToContextConnection::ScriptContextFailedToStart { jobDataIdentifier, serviceWorkerIdentifier, exceptionMessage }, 0);
+    Ref protectedConnection = m_connectionToNetworkProcess;
```
ditto.

cdumez commented on Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
```diff
@@ -269,7 +274,8 @@ void WebSWContextManagerConnection::startFetch(SWServerConnectionIdentifier serv
     auto serviceWorkerThreadProxy = SWContextManager::singleton().serviceWorkerThreadProxyFromBackgroundThread(serviceWorkerIdentifier);
     if (!serviceWorkerThreadProxy) {
-        m_connectionToNetworkProcess->send(Messages::ServiceWorkerFetchTask::DidNotHandle { }, fetchIdentifier);
+        Ref protectedConnection = m_connectionToNetworkProcess;
```
ditto. Please apply elsewhere in this file. I'll stop commenting on these.

cdumez commented on Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
```diff
@@ -582,22 +602,26 @@ bool WebSWContextManagerConnection::isThrottleable() const
 void WebSWContextManagerConnection::removeNavigationFetch(WebCore::SWServerConnectionIdentifier serverConnectionIdentifier, WebCore::FetchIdentifier fetchIdentifier)
 {
-    m_queue->dispatch([protectedThis = Ref { *this }, serverConnectionIdentifier, fetchIdentifier] {
+    Ref protectedQueue = m_queue;
```
`m_queue` can be `const` and then we won't need the local Ref.

cdumez reviewed (approved):
LGTM

kmonsen committed:
Address Safer CPP warnings in WebSWContextManagerConnection.cpp : updateAppInitiatedValue
https://bugs.webkit.org/show_bug.cgi?id=289629
rdar://146879414

Reviewed by Chris Dumez.

Making member refs const when they can be.

* Source/WebKit/SaferCPPExpectations/UncheckedLocalVarsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedLocalVarsCheckerExpectations:
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.cpp:
(WebKit::WebSWContextManagerConnection::updateAppInitiatedValue):
(WebKit::WebSWContextManagerConnection::installServiceWorker):
* Source/WebKit/WebProcess/Storage/WebSWContextManagerConnection.h:

Canonical link: https://commits.webkit.org/292294@main
